### PR TITLE
idefics2 enable_input_require_grads not aligned with disable_input_re…

### PR DESCRIPTION
**Pull request by sywangyi**
…quire_grads

make peft+idefics2 checkpoints disable fail

error like
AttributeError: 'GaudiIdefics2ForConditionalGeneration' object has no attribute '_require_grads_hook'

- vision models: @amyeroberts

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for opening this PR and fixing! 

For my own understanding of the changes - it seems there's two changes happening here: one which ensures proper disabling of hooks, the other which adds peft-related logic to the tests. Is the latter necessary to catch the former?

**Comment by sywangyi** [comment]
> Thanks for opening this PR and fixing!
> 
> For my own understanding of the changes - it seems there's two changes happening here: one which ensures proper disabling of hooks, the other which adds peft-related logic to the tests. Is the latter necessary to catch the former?

yes, the latter is used to catch the former.

**Comment by amyeroberts** [comment]
@sywangyi Could you explain a bit why? In the case of the test, `_hf_peft_config_loaded=False` seems to be testing the previous logic (and I assume passing on main). The reason I ask is that it's good we find a way to catch possible issues but the relation to PEFT isn't obvious to me -- this indicates that either the original test isn't properly capturing what we thought or there's a different behaviour in the case of peft which e might need to address. 

In particular, I think that we probably want a peft-specific test to be separated out of the current test to make sure what's being tested and why is clear

**Comment by sywangyi** [comment]
see logic in https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2380 and https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L2429 this path (_hf_peft_config_loaded=True) is not tested. so I set _hf_peft_config_loaded to False and True to test both path.

**Comment by amyeroberts** [comment]
@sywangyi OK, I see. Hmmm -- we shouldn't really need to know about PEFT + trainer within the modeling utils but alas it's there. Could you split up the tests so that we have a separate one for the PEFT condition and resolve the failing tests? After than I think the PR will be good to go

**Comment by sywangyi** [comment]
I check the failure case
FAILED examples/tensorflow/test_tensorflow_examples.py::ExamplesTests::test_run_image_classification - ValueError: Task image-classification is not compatible with this dataset! Available tasks: []
======== 1 failed, 5 passed, 3 skipped, 8 warnings in 300.61s (0:05:00) ========

this case has nothing to do with the PR

File: src/transformers/modeling_utils.py
Comment (amyeroberts) [inline src/transformers/modeling_utils.py:1768]: `get_input_embeddings` shouldn't be `None` c.f. the docstring. What tests does this address?

Comment (sywangyi) [reply src/transformers/modeling_utils.py:1768]: https://github.com/huggingface/transformers/blob/main/src/transformers/models/speecht5/modeling_speecht5.py#L2111
speecht5 case failure since it may return None

Comment (sywangyi) [reply src/transformers/modeling_utils.py:1768]: FAILED tests/models/speecht5/test_modeling_speecht5.py::SpeechT5ModelTest::test_peft_gradient_checkpointing_enable_disable - AttributeError: 'NoneType' object has no attribute 'register_forward_hook'
FAILED tests/models/speecht5/test_modeling_speecht5.py::SpeechT5ForSpeechToSpeechTest::test_peft_gradient_checkpointing_enable_disable - AttributeError: 'NoneType' object has no attribute 'register_forward_hook'

Comment (amyeroberts) [reply src/transformers/modeling_utils.py:1768]: OK, this seems to be a speech t5 specific issue and sohuld be resolved but I wouldn't want this to block this PR. Could you: 

* Remove the `is not None` condition here
* Open an issue, linking to this PR and include the info you included here 
* Skip the test for SpeechT5 + add a link to the github issue in the skip message

Once done we can merge :) 

cc @ylacombe

Comment (sywangyi) [reply src/transformers/modeling_utils.py]: done

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing and iterating on this!

**Pull request merged**
